docs/programming-guide.md

```diff
@@ -1144,9 +1144,11 @@ generate these on the reduce side. When data does not fit in memory Spark will s
 to disk, incurring the additional overhead of disk I/O and increased garbage collection.
 
 Shuffle also generates a large number of intermediate files on disk. As of Spark 1.3, these files
```

Comment:
I know this has been merged, but a annoying issue that I have found in docs (including mine, so I am guilty too) is use of this `as of Spark X`. No one remembers searching for this pattern and it never gets updated. Rather we should use markdown variables, `as of Spark {{site.SPARK_VERSION_SHORT}}`.